docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,73 @@
+The [ACL plugin](https://github.com/fabric8io/openshift-elasticsearch-plugin)<sup>[2](#2)</sup> that is deployed
+as part of aggregated logging will be updated to allow access from the gateway.  ElasicSearch index management<sup>[3](#3)</sup>  will be modified so
+that it will not cull APIMan policy data.
+
+#### Off-Cluster Policy Storage
+Cluster administrators can configure APIMan to use an alternate ElasticSearch cluster if desired.  Some organizations may have an existing ElasticSearch cluster that already defines
+policy, and adminstrators may desire to use it instead of the one provided with aggregated logging.  It is assumed in this configuration, the APIMan management interface will
+additionally rely on Kibana deployed off-cluster.  The multi-tenant features provided by aggregated logging for ElasticSearch and Kibana may not be available when configured to utilize these off-cluster
+dependencies.      
+
+## Use Cases
+* **UC01** As an API provider, I want to navigate from the OpenShift web console to the policy management interface, so I can define my service policies.
+* **UC02** As an API provider, I want to associate an APIMan organization with an OpenShift project, so I can control policy between organizations.
+* **UC03** As an API provider, I want to expose my service, so that it is consumable according to my management policy.  
+* **UC04** As a cluster administrator, I want a gateway to route API traffic only, so that it manages traffic based on a service's policy.
+* **UC05** As a cluster administrator, I want to deploy APIMan and its components reusing existing infrastructure components where possible (e.g. ElasticSearch), so I can minimize infrastructure components.
+* **UC06** As a cluster administrator, I want API management interface to be able to discover the route to Kibana, so it can be linked to from the management interface. 
+* **UC07** As an API consumer, I want to configure my services to consume other services
+
+## Specification
+
+## Project Annotations
+add annotation keys an sample here
+## Service Annotations (and labels?)
+add sample here to upstream work
+do we need labels to generate backlink from apiman management to console service?
+
+### Origin Web Console UI extension
+UI exension that creates link to apiman management with context based on project/service annotations
+
+### OpenShift CLI Modifications
+service annotation command? 
+
+### Deployment / Deployer pod
+
+## Rationale
+The technical rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other products.
+The rationale should provide evidence of consensus within the community and discuss important objections or concerns raised during discussion.
+
+OpenShift existing arch allows for deploying additional routers 
+
+## Limitations
+scalability - 
```

Comment:
Some initial comments to get going:

> +### Origin Web Console UI extension
> +UI exension that creates link to apiman management with context based on project/service annotations
We should add SSO support for this. I would prefer if the SSO can be done cookie based, so as long you have session, a link into the apiman console would work. At the moment you need a 'special link' with an auth-token on it and lots of links to the apiman console don't have the token: it's very brittle, plus token expiration is a problem. Maybe it's less of an issue if the apiman-console can run in the same domain as the openshift console.

> +### OpenShift CLI Modifications
> +service annotation command? 
Should the integration should support  create, view, and delete 'apiman-routes'? I think it'd need to support at least view and delete.

> ## Limitations
> +scalability - 
HTTP traffic only (for now).
If apiman is one per cluster the service catalog displays all services on the cluster. A feature request is submitted to Apiman to support 'private' services and or namespaces. Having such a 'cluster wide' catalog can be seen as an asset too. 

## Users and Roles
How do openshift users and roles translate to